support/CMakeLists.txt

```
@@ -0,0 +1,29 @@
+#
+# Copyright (c) 2021-present, Trail of Bits, Inc.
+# All rights reserved.
+#
+# This source code is licensed in accordance with the terms specified in
+# the LICENSE file found in the root directory of this source tree.
+#
+
+# Substitute the build and install directories into the support helpers
+configure_file(
+  "${CMAKE_CURRENT_LIST_DIR}/Support.cpp.in"
+  "${CMAKE_CURRENT_BINARY_DIR}/Support.cpp"
+  @ONLY
+)
+
+set(sleigh_support_source_list
+  "${CMAKE_CURRENT_BINARY_DIR}/Support.cpp"
+)
+
+add_library(support
+  ${sleigh_support_source_list}
+)
+
+add_library(sleigh::support ALIAS support)
+
+target_link_libraries(support PRIVATE
+  sleigh::sleigh_settings
+)
```

I forget the exact reason why the other libraries link the settings as `PUBLIC`, but I think we should be consistent and do it here too.

```suggestion
target_link_libraries(support PUBLIC
  sleigh::sleigh_settings
)
```

Side-note: the `sleigh_settings` target is really weird and I think we should remove the target in favor of CMake's [`set_target_properties`](https://cmake.org/cmake/help/latest/command/set_target_properties.html) where we can do pretty much the same thing without having to link to this weird settings pseudo-library and worry about correctly propagating the settings.